fitparser/src/de/parser.rs

```diff
@@ -36,9 +37,23 @@ impl Value {
             Value::SInt64(val) => *val != 0x7FFF_FFFF_FFFF_FFFF,
             Value::UInt64(val) => *val != 0xFFFF_FFFF_FFFF_FFFF,
             Value::UInt64z(val) => *val != 0x0,
-            // TODO: I need to check this logic, since for Byte Arrays it's only invalid if
-            // all the values are invalid. Is that the case for all array fields or just "byte arrays"?
-            Value::Array(vals) => !vals.is_empty() && vals.iter().any(|v| v.is_valid()),
+            Value::Array(vals) => {
+                if let Some(first) = vals
+                    .iter()
+                    .find(|&v| !matches!(v, Value::Invalid))
+                    .map(discriminant)
```

Comment:
This PR looks great, however, I would like to drop this check on whether or not the array is homogeneous or not. Theoretically it's a valid check due to how the Value enum is defined, but in actuality it's somewhat complicated dead code since there's no way for a mixed array to be parsed. The `data_field_value` function which builds any Value::Array instances in this module operates on a fixed base_type provided as a function argument (other than `Value::Invalid` I suppose).